vscode/src/copilot/webview/hljs-qsharp.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { HLJSApi, Mode, type Language } from "highlight.js";
+
+// NOTE: This has a bunch of differences from how we structure the language in <vscode/syntaxes/qsharp.tmLanguage.json>
+// We should align with that as much as possible eventually.
+
+const KEYWORDS_GENERAL = [
+  "use",
+  "borrow",
+  "mutable",
+  "let",
+  "set",
+  "if",
+  "elif",
+  "else",
+  "repeat",
+  "until",
+  "fixup",
+  "for",
+  "in",
+  "while",
+  "return",
+  "fail",
+  "within",
+  "apply",
+];
+
+const KEYWORDS_DECL = [
+  "namespace",
+  "open",
+  "import",
+  "export",
+  "as",
+  "internal",
+  "newtype",
+  "struct",
+  "operation",
+  "function",
+  "new",
+  "body",
+  "adjoint",
+  "Adjoint",
+  "controlled",
+  "Controlled",
+  "self",
+  "auto",
+  "distribute",
+  "invert",
+  "intrinsic",
+];
+
+const KEYWORDS_TYPE = [
+  "Int",
+  "BigInt",
+  "Double",
+  "Bool",
+  "Qubit",
+  "Pauli",
+  "Result",
+  "Range",
+  "String",
+  "Unit",
+  "Ctl",
+  "Adj",
+  "is",
+];
+
+const KEYWORDS_CONSTANTS = [
+  "true",
+  "false",
+  "PauliI",
+  "PauliX",
+  "PauliY",
+  "PauliZ",
+  "One",
+  "Zero",
+];
+
+export default function (hljs: HLJSApi): Language {
+  const QSHARP_KEYWORDS = {
+    keyword: KEYWORDS_GENERAL,
+    declare: KEYWORDS_DECL,
+    literal: KEYWORDS_CONSTANTS,
+    type: KEYWORDS_TYPE,
+  };
+
+  const QSHARP_OPERATOR_MODE: Mode = {
+    scope: "operator",
+    match:
+      "\\b(not|and|or)\\b|\\b(w/)|(=)|(!)|(<)|(>)|(\\+)|(-)|(\\*)|(\\/)|(\\^)|(%)|(\\|)|(\\&\\&\\&)|(\\~\\~\\~)|(\\.\\.\\.)|(\\.\\.)|(\\?)",
+  };
+
+  const QSHARP_NUMBER_MODE: Mode = {
+    scope: "number",
+    begin: "\\b[\\d_]*\\.?[\\d_]\\b",
+  };
+
+  const QSHARP_FN_CALL: Mode = {
+    scope: "title",
+    match: /\b[a-zA-Z_][a-zA-Z0-9_]*\s*(?=\()/,
+  };
+
+  const QSHARP_PUNCTUATION = {
+    match: /[;:(){},]/,
+    className: "punctuation",
+    relevance: 0,
+  };
+
+  const QSHARP_INTERP_STRING_MODE: Mode = {
+    scope: "string",
+    begin: /\$"/,
+    end: '"',
+    contains: [
+      {
+        scope: "subst",
+        begin: /\{/,
+        end: /\}/,
+        keywords: QSHARP_KEYWORDS,
+        contains: [
+          /* add later */
```

Comment:
Looks like this is missing the list of modes (should be largely a copy of the list below, except with `QSHARP_INTERP_STRING_MODE` replaced by `"self"`)

Reply:
Exactly, that's why I decided against going down the rabbit hole. Once you decide support certain things, you have to support expressions overall.

I felt that it interpolated strings are important since the samples contain quite a few. They never contain anything more than a single identifier though. Can't we just live with it?

Reply:
Think the latest change fixes it:

<img width="440" alt="image" src="https://github.com/user-attachments/assets/02ddc689-b809-4a7d-bdae-926110a29fe1" />